tools/src/main/scala/scala/scalanative/interflow/MergeProcessor.scala

```diff
@@ -150,7 +151,7 @@ final class MergeProcessor(
           def escapes(addr: Addr): Boolean =
             states.exists(_.hasEscaped(addr))
 
-          states.head.heap.keys
+          states.head.heap.keys.iterator
```

Comment:
Wouldn't it be preferable to call `keysIterator` directly?

```suggestion
          states.head.heap.keysIterator
```